cert.go

```diff
@@ -413,3 +421,17 @@ func (c *Certificate) SetVersion(version X509_Version) error {
 	}
 	return nil
 }
+
+// GetExtensionValue returns the value of the given NID's extension.
+func (c *Certificate) GetExtensionValue(nid NID) []byte {
+	dataLength := C.int(0)
+	val := C.get_extention(c.x, C.int(nid), &dataLength)
+	return charToBytes(val, int(dataLength))
+}
+
+// charToBytes converts c unisgned char to golang bytes
+func charToBytes(src *C.uchar, sz int) []byte {
+	dest := make([]byte, sz)
+	copy(dest, (*(*[1024]byte)(unsafe.Pointer(src)))[:sz:sz])
```

Comment:
I don't think this is safe. This could cause us to read outside of a mapped page.

I _think_ we need to call `C.GoBytes` (and maybe copy it? I'm not sure).

Reply:
Now, I'm using GoBytes.
Thanks for the suggestion.